### PR TITLE
API Portal React Query POC. ENG-337

### PR DESCRIPTION
**Pull request by rdgfuentes**
### Substituting Redux with ReactQuery
- Migrated redux store, slices and hooks to ReactQuery query client hooks and mutation hooks
- Added caching, memoizing, deduping, updating out of date (sync) data with ReactQuery
- Implemented authentication flow with ReactQuery
- Removed unnecessary dependencies

### CODE REVIEW

File: apps/api-portal/features/user-applications/user-applications.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+async function deleteUserApplication(id: number) {
+  const url = `/v2/auth/user-applications/${id}`
+  const userApplication = await GFWAPI.fetch<UserApplication>(url, {
+    method: 'DELETE',
+    responseType: 'default',
+  })
+  return { ...userApplication, id }
+}
+
+export const useDeleteUserApplication = (userId) => {
```
Comment (j8seangel) [inline apps/api-portal/features/user-applications/user-applications.ts:44]: what about decoupling this from the component to its own hook file? 
I think that is easier to reuse and to read

File: apps/api-portal/package.json [hunk 2]
```diff
@@ -27,11 +26,8 @@
     "react-ga": "3.3.0",
     "react-i18next": "11.14.3",
     "react-map-gl": "^5.3.17",
-    "react-redux": "^7.2.6",
```
Comment (j8seangel) [inline apps/api-portal/package.json:30]: Thanks!

**Review by j8seangel — APPROVED** [review]
Great work, it looks good to me! Looking forward to know your thoughts about the library.
The main difference I see is the debug tool which I think it is great but let's talk in net frontend meeting

File: apps/api-portal/components/user-additional-fields/user-additional-fields.tsx [hunk 3]
```diff
@@ -57,7 +88,12 @@ export function UserAdditionalFields(props: UserAdditionalFieldsProps) {
     () => !!userAdditionalInformation?.apiTerms,
     [userAdditionalInformation?.apiTerms]
   )
-  if (loading) return <Spinner></Spinner>
+
+  useEffect(() => {
+    isSuccess && router.replace('/')
```
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply apps/api-portal/components/user-additional-fields/user-additional-fields.tsx:93]: I like the idea, thanks!

**Pull request merged**